### PR TITLE
Visualization H2

### PR DESCRIPTION
**Pull request by AliceLeBrigant**
In ```visualization``` for ```space=H2_poincare_half_plane```,
- adapt the axis bounds of the Poincare half plane plot to the points that are plotted
- authorize points to be given in Poincare half plane coordinates, and make it the default representation, instead of the extrinsic coordinates. I would be in favour of doing the same for the other representations, as I think it is more natural for the user to give the points' coordinates in the appropriate representation.

### CODE REVIEW

File: geomstats/visualization.py [hunk 1]
```diff
@@ -342,28 +343,41 @@ def draw(self, ax, **kwargs):
...
+        xmin = points[gs.argmin(points[:, 0]), 0]
+        xmax = points[gs.argmax(points[:, 0]), 0]
+        ymax = points[gs.argmax(points[:, 1]), 1]
+        xmargin = 0.1 * (xmax - xmin)
+        ymargin = 0.1 * ymax
         plt.setp(ax,
-                 xlim=(-ax_s, ax_s),
-                 ylim=(0., ax_s),
+                 xlim=(xmin - xmargin, xmax + xmargin),
+                 ylim=(0., ymax + ymargin),
```
Comment (nguigs) [inline geomstats/visualization.py:380]: Is this necessary to manually set xlim/ylim? It is the only thing preventing the exemple from running with tensorflow, and matplotlib does a good job at defining the limits automatically!

File: geomstats/visualization.py [hunk 2]
```diff
@@ -486,11 +500,25 @@ def convert_to_trihedron(point, space=None):
...
+         point_type=None, **point_draw_kwargs):
+    """Plot points in one of the implemented manifolds.
+
+    The implemented manifolds are:
+    - the special orthogonal group SO(3)
+    - the special Euclidean group SE(3)
+    - the circle S1 and the sphere S2
+    - the hyperbolic plane (the Poincare disk, the Poincare
+      half plane and the Klein disk)
+    - the Poincare polydisk
```
Comment (nguigs) [inline geomstats/visualization.py:512]: Much nicer!

File: tests/test_examples.py [hunk 3]
```diff
@@ -117,6 +117,7 @@ def test_plot_square_h2_square_poincare_disk():
         plot_square_h2_poincare_disk.main()
 
     @staticmethod
+    @geomstats.tests.np_and_pytorch_only
```
Comment (nguigs) [inline tests/test_examples.py:120]: Could be removed, see above comment

File: geomstats/visualization.py [hunk 4]
```diff
@@ -342,28 +343,41 @@ def draw(self, ax, **kwargs):
...
-        points = self.convert_to_half_plane_coordinates(points)
+        if self.point_type == 'extrinsic':
+            if not gs.all(H2.belongs(points)):
+                raise ValueError(
+                    'Points do not belong to the hyperbolic space '
+                    '(extrinsic coordinates)')
+            points = self.convert_to_half_plane_coordinates(points)
+        elif self.point_type == 'half-plane':
+            if not gs.all(POINCARE_HALF_PLANE.belongs(points)):
+                raise ValueError(
```
Comment (nguigs) [inline geomstats/visualization.py:361]: These lines are not covered by the unit-tests, maybe add a test with point_type=half-plane?

File: tests/test_visualization.py [hunk 5]
```diff
@@ -56,7 +56,8 @@ def test_plot_points_h2_poincare_disk(self):
     @geomstats.tests.np_only
     def test_plot_points_h2_poincare_half_plane(self):
         points = self.H2.random_uniform(self.n_samples)
-        visualization.plot(points, space='H2_poincare_half_plane')
+        visualization.plot(points, space='H2_poincare_half_plane',
+                           point_type='extrinsic')
```
Comment (nguigs) [inline tests/test_visualization.py:60]: Also test `point_type=None` to improve coverage

**Review by nguigs — COMMENTED** [review]
Great, I agree with you, its nicer to plot points in the half-plane coordinate system straight away! Do we want to have `point_type='half-plane'` or `'half-space'` to be consistent with the hyperbolic class?

File: geomstats/visualization.py (hunk 4 above)
Comment (nguigs) [reply geomstats/visualization.py:361]: (Have a look at https://sourcegraph.com/github.com/geomstats/geomstats to find out which lines are not being tested)

File: geomstats/visualization.py (hunk 1 above)
Comment (AliceLeBrigant) [reply geomstats/visualization.py]: Good point !

**Comment by AliceLeBrigant** [comment]
> Great, I agree with you, its nicer to plot points in the half-plane coordinate system straight away! Do we want to have `point_type='half-plane'` or `'half-space'` to be consistent with the hyperbolic class?

Probably 'half-space' is better. Thanks for the comments !
